cas-server-webapp-config/src/main/java/org/apereo/cas/config/CasAuditTrailConfiguration.java

```diff
@@ -194,5 +187,16 @@ public Map auditResourceResolverMap() {
         map.put("SAVE_SERVICE_RESOURCE_RESOLVER", returnValueResourceResolver());
         return map;
     }
+
+    /**
+     * Principal resolver.
+     *
+     * @param centralAuthenticationService
+     * @return the principal resolver
+     */
+    @Bean(name = "auditablePrincipalResolver")
+    public PrincipalResolver auditablePrincipalResolver(CentralAuthenticationService centralAuthenticationService) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Parameter centralAuthenticationService should be final.](https://www.codacy.com/app/mmoayyed/cas/file/2639442193/issues/source?bid=3256115&fileBranchId=3321105#l198)